test/mypy_visitor.py

```diff
@@ -0,0 +1,191 @@
+
+So roughly, we do this:
+
+1. Import the mypy things we need
+2. Capture the globals (so that we can resolve the strigified type annotations
+   to the correct types)
+3. Clear the mypy imported modules
+4. Import them again with their pure python versions
+5. Inspect the Visitor to get the type names, but resolve them using the
+   captured globals (from the native versions)
+6. Restore the native mypy implementations
+
+"""
+import inspect
+import sys
+import typing
+from collections.abc import Iterator
+from contextlib import contextmanager
+from dataclasses import dataclass
+from importlib.abc import PathEntryFinder
+from importlib.machinery import FileFinder
+from types import FunctionType
+from typing import Any, Callable
+
+import mypy.nodes
+import mypy.traverser
+import pytest
+
+VisitorNodeTypeMap = dict[str, mypy.nodes.Node]
+Namespace = dict[str, Any]  # type: ignore
+
+
+@contextmanager
+def prefer_pure_python_imports() -> Iterator[None]:
+    """
+    During the scope of this context manager, all imports will be done using
+    pure python versions when available.
+
+    Credit to this answer on SO: https://stackoverflow.com/a/68685189/
+    """
+
+    @dataclass
+    class PreferPureLoaderHook:
+        orig_hook: Callable[[str], PathEntryFinder]
+
+        def __call__(self, path: str) -> PathEntryFinder:
+            finder = self.orig_hook(path)
+            if isinstance(finder, FileFinder):
+                # Move pure python file loaders to the front
+                finder._loaders.sort(  # type: ignore
+                    key=lambda pair: 0 if pair[0] in (".py", ".pyc") else 1
+                )
+
+            return finder
+
+    sys.path_hooks = [PreferPureLoaderHook(h) for h in sys.path_hooks]
+    sys.path_importer_cache.clear()
+
+    yield
+
+    # Restore the previous behaviour
+    original_hooks = []
+    for hook in sys.path_hooks:
+        assert isinstance(hook, PreferPureLoaderHook)
+        original_hooks.append(hook.orig_hook)
+    sys.path_hooks = original_hooks
+    sys.path_importer_cache.clear()
+
+
+@contextmanager
+def pure_python_mypy() -> Iterator[None]:
+    """
+    Inside this context, all mypy related imports are done with the pure python
+    versions.
+
+    Any existing mypy module that was imported before needs to be reimported
+    before use within the context.
+
+    Upon exiting, the previous implementations are restored.
+    """
+
+    def loaded_mypy_modules() -> Iterator[str]:
+        """Covenient block to get names of imported mypy modules"""
+        for mod_name in sys.modules:
+            if mod_name == "mypy" or mod_name.startswith("mypy."):
+                yield mod_name
+
+    # First, backup all imported mypy modules and remove them from sys.modules,
+    # so they will not be found in resolution
+    saved_mypy = dict()
+    for mod_name in list(loaded_mypy_modules()):
+        saved_mypy[mod_name] = sys.modules[mod_name]
+        del sys.modules[mod_name]
```

Comment:
This could be written like this:

```python
saved_mypy[mod_name] = sys.modules.pop(mod_name)
```

The "no del" check (FURB131) doesn't currently catch this because of the `.` part.